tests/ext-monitor.test.js

```diff
@@ -78,24 +94,85 @@ describe('start extension monitoring', () => {
       'EAM is already running'
     );
   });
+
+  test('start monitoring newly installed extensions automatically', async () => {
+    const onExtActivityAddListener = jest.fn();
+    const onInstalledAddListener = jest.fn();
+    const onUninstalledAddListener = jest.fn();
+
+    const selfExt = { id: 'ext1', type: 'extension' };
+    const initialMonitoringExts = [
+      { id: 'ext2', type: 'extension' },
+      { id: 'ext3', type: 'extension' },
+    ];
+    const newExtension = { id: 'ext4', type: 'extension' };
+    const newTheme = { id: 'ext5', type: 'theme' };
+    const expectMonitorExtIds = ['ext2', 'ext3', 'ext4'];
+
+    window.browser = {
+      activityLog: {
+        onExtensionActivity: { addListener: onExtActivityAddListener },
+      },
+      management: {
+        onInstalled: { addListener: onInstalledAddListener },
+        onUninstalled: { addListener: onUninstalledAddListener },
+      },
+      runtime: { id: selfExt.id },
+    };
+
+    const extMonitor = new ExtensionMonitor();
+    const getAllExtensionsFn = jest.spyOn(extMonitor, 'getAllExtensions');
+
+    extMonitor.selfId = selfExt.id;
```

Comment:
`selfId` is not used any more. Why is it here?

Reply:
sorry, didn't catch my eye. Thanks for noticing. Removed now :-)